crates/c/src/lib.rs

```diff
@@ -471,46 +468,51 @@ impl C {
     /// This function will trim the sets on `self` to only retain those types
     /// which exports refer to that come from imports.
     fn remove_types_redefined_by_exports(&mut self, resolve: &Resolve, world: WorldId) {
-        // First build up a set of all types used by exports and all the
-        // exported interfaces.
-        let mut live_export_types = LiveTypes::default();
-        let mut exported_interfaces = HashSet::new();
-        for (_, export) in resolve.worlds[world].exports.iter() {
-            match export {
-                WorldItem::Function(_) => {}
-                WorldItem::Interface(i) => {
-                    exported_interfaces.insert(*i);
-                    live_export_types.add_interface(resolve, *i)
-                }
-                WorldItem::Type(_) => unreachable!(),
+        let live_import_types = find_live_export_types(resolve, world);
+        self.dtor_funcs.retain(|k, _| live_import_types.contains(k));
+        self.type_names.retain(|k, _| live_import_types.contains(k));
+        self.resources.retain(|k, _| live_import_types.contains(k));
+    }
+}
+
+pub fn find_live_export_types(resolve: &Resolve, world: WorldId) -> HashSet<TypeId> {
```

Comment:
The name here isn't quite what this function is returning, perhaps something like `imported_types_used_by_exported_interfaces` or something like that?